### PR TITLE
Bump google.golang.org/grpc to latest 

### PR DESCRIPTION
**Pull request by winkingturtle-vmw**
Summary
---------------
Bump google.golang.org/grpc to latest to address CVE: GHSA-m425-mq94-257g CVE-2023-44487

- Newer grpc throws an error when address is not defined, so tests were changed to reflect the new changes
- Remove the use of deprecated `grpc.WithInsecure()` with the correct package

### CODE REVIEW

File: src/code.cloudfoundry.org/bosh-dns-adapter/copilot/client.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func NewConnectedClient(serverAddr string, dialOpts ...DialOption) (*Client, err
 	}
 
 	if opts.withInsecure {
-		grpcOpts = append(grpcOpts, grpc.WithInsecure())
+		grpcOpts = append(grpcOpts, grpc.WithTransportCredentials(insecure.NewCredentials()))
```
Comment (MarcPaquette) [inline src/code.cloudfoundry.org/bosh-dns-adapter/copilot/client.go:38]: Why are we using insecure here?

File: src/code.cloudfoundry.org/bosh-dns-adapter/main_test.go [hunk 2]
```diff
@@ -184,17 +185,16 @@ var _ = Describe("Main", func() {
...
 			url := fmt.Sprintf("http://127.0.0.1:%s?type=1&name=app-id.istio.local.", dnsAdapterPort)
 			request, err := http.NewRequest("GET", url, reader)
 			Expect(err).To(Succeed())
 
-			resp, err := http.DefaultClient.Do(request)
-			Expect(err).To(Succeed())
+			_, err = http.DefaultClient.Do(request)
+			Expect(err).NotTo(Succeed())
 
-			Expect(resp.StatusCode).To(Equal(http.StatusInternalServerError))
```
Comment (MarcPaquette) [inline src/code.cloudfoundry.org/bosh-dns-adapter/main_test.go:197]: Response is no longer needed?

**Review by MarcPaquette — COMMENTED** [review]
Just need a couple of clarifications on the code changes.

File: src/code.cloudfoundry.org/bosh-dns-adapter/copilot/client.go (hunk 1 above)
Comment (winkingturtle-vmw) [reply src/code.cloudfoundry.org/bosh-dns-adapter/copilot/client.go:38]: I think historically, the team added the insecure option for the tests. I don't think that's what we use in production.

File: src/code.cloudfoundry.org/bosh-dns-adapter/main_test.go (hunk 2 above)
Comment (winkingturtle-vmw) [reply src/code.cloudfoundry.org/bosh-dns-adapter/main_test.go:197]: Yes, it errors out before setting a `resp` object

**Review by MarcPaquette — APPROVED** [review]

**Pull request merged**